x/trigger/keeper/msg_server.go

```diff
@@ -0,0 +1,71 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the account MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+// CreateTrigger creates new trigger from msg
+func (s msgServer) CreateTrigger(goCtx context.Context, msg *types.MsgCreateTriggerRequest) (*types.MsgCreateTriggerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	event, err := msg.GetTriggerEventI()
+	if err != nil {
+		return nil, err
+	}
+	if err = event.ValidateContext(ctx); err != nil {
+		return nil, err
+	}
+
+	trigger := s.NewTriggerWithID(ctx, msg.GetAuthority(), msg.GetEvent(), msg.GetActions())
+	s.RegisterTrigger(ctx, trigger)
+
+	err = ctx.EventManager().EmitTypedEvent(&types.EventTriggerCreated{
+		TriggerId: fmt.Sprintf("%d", trigger.GetId()),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.MsgCreateTriggerResponse{Id: trigger.GetId()}, nil
+}
+
+// DestroyTrigger destroys trigger from msg
+func (s msgServer) DestroyTrigger(goCtx context.Context, msg *types.MsgDestroyTriggerRequest) (*types.MsgDestroyTriggerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	trigger, err := s.GetTrigger(ctx, msg.GetId())
+	if err != nil {
+		return nil, err
+	}
+	if trigger.GetOwner() != msg.GetAuthority() {
+		return nil, types.ErrInvalidTriggerAuthority
+	}
```

Comment:
Probably a larger discussion point here. Should we allow triggers to be destroyed via governance proposal?

I guess that'd be pretty easy to add in a later iteration if we decide it'd be desirable.

Reply:
I think that's a good idea, but I agree we probably want to discuss it first.